api_tests/src/comment.spec.ts

```diff
@@ -112,22 +112,59 @@ test("Update a comment", async () => {
 });
 
 test("Delete a comment", async () => {
+  // creating a comment on alpha (remote from home of community)
   let commentRes = await createComment(alpha, postRes.post_view.post.id);
 
+  // Find the comment on beta (home of community)
+  let betaComment = (
+    await resolveComment(beta, commentRes.comment_view.comment)
+  ).comment;
+
+  if (!betaComment) {
+    throw "Missing beta comment before delete";
+  }
+
+  // Find the comment on gamma
+  // This is testing replication from remote-home-remote (alpha-beta-gamma)
+  let gammaComment = (
+    await resolveComment(gamma, commentRes.comment_view.comment)
+  ).comment;
+
+  if (!gammaComment) {
+    throw "Missing gamma comment (remote-home-remote replication) before delete";
+  }
+
   let deleteCommentRes = await deleteComment(
     alpha,
     true,
     commentRes.comment_view.comment.id,
   );
   expect(deleteCommentRes.comment_view.comment.deleted).toBe(true);
 
-  // Make sure that comment is undefined on beta
+  // Make sure that comment is undefined on beta after delete
   let betaCommentRes = (await resolveComment(
     beta,
     commentRes.comment_view.comment,
   )) as any;
+  if (betaCommentRes) {
+    console.log(betaCommentRes.comment);
+  }
   expect(betaCommentRes.error).toBe("couldnt_find_object");
 
+  // Make sure that comment is undefined on gamma after delete
+  // This is testing replication from remote-home-remote (alpha-beta-gamma)
+  let gammaCommentRes = (await resolveComment(
+    gamma,
+    commentRes.comment_view.comment,
+  )) as any;
+  if (gammaCommentRes) {
+    if (gammaCommentRes.comment) {
+      console.log("gamma has deleted comment", gammaCommentRes.comment.comment);
+    }
+  }
+  expect(gammaCommentRes.error).toBe("couldnt_find_object");
```

Comment:
Okay turns out its like I mentioned in my other comment, the problem is in activity_lists.rs. Removing the line `PersonInboxActivities::Delete` fixes this test. So it enters the wrong variant which doesnt forward it to followers. Have to think about the proper way to fix it, probably some rewrite of that file is necessary.

Reply:
Any idea if what is going on with my revisions to "Report a comment" test? Did I make a mistake?

Reply:
FYI the way to debug this is by looking at the Lemmy logs in `less /tmp/lemmy_beta.out` etc, and adding `dbg!` statements in the Rust code to see what it is doing.